### PR TITLE
Ensure that we use rooted paths for adding/removing source files.

### LINKED ISSUE #986: VS2017 RC2 error list shows netstandard library project compiles all files twice
Using d15rel 26014.00.

```
git clone -b vs2017MT https://github.com/AArnott/Validation.git
cd Validation
git checkout 6045abdc
src\Validation.sln
```

The error list in VS2017 RC2 shows many errors around ambiguous calls because the language service is compiling each file twice. Yet when I build the project, the output window clearly shows that problem is not in the build itself. (A few errors are there, but they are unrelated).

Error	CS0121	The call is ambiguous between the following methods or properties: ...

image

### LINKED ISSUE #1030: Ambiguity between same member with linked source file.
Reported through [VS Feedback](https://developercommunity.visualstudio.com/content/problem/2482/ambiguity-between-same-member-with-linked-source-f.html)

If a source file is added as a link in a .Net Standard library, all references to members inside the source file (class source file) all have the issue of "Ambiguity between 'Class.Member' and 'Class.Member'". If the member definition name is changed, and the source file is saved, the error disappears and the error that no member with that name exists appears (meaning that there is only a single definition of the member in the whole project). If the same file is load in via normal copying, the error doesn't occur. 

Project that shows the repro is [here](https://developercommunity.visualstudio.com/storage/attachments/373-20161120ambiguity.zip)

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #986 and Fixes #1030

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/project-system @davkean for review - this fixes the duplicate source file ambiguity errors.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs [hunk 1]
```diff
@@ -121,6 +121,8 @@ public override Task OnContextReleasedAsync(IWorkspaceProjectContext context)
 
         private void RemoveSourceFile(string fullPath, IWorkspaceProjectContext context)
         {
+            fullPath = _project.MakeRooted(fullPath);
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: Add a test for this?

Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: Do we have LS tests?

Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: Let me try add a simple test for SourceItemHandler

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: Can we please unify this? We're already making rooted in the evaluation case. Can you remove those cases?

Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: Done.

Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: @srivatsn I figured out I cannot still add unit tests for `ILanguageServiceCommandLineHandler` as the Handle method takes `IWorkspaceProjectContext` which doesn't yet give IVT access to our unit test projects.

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: Did I not submit that code review?

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: https://github.com/dotnet/roslyn/pull/13763

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: This was merged in.

Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: Seems like we are still building with a very old version MS.VS.LS NuGet package: https://github.com/dotnet/roslyn-project-system/blob/master/src/Dependencies/Roslyn/project.json#L4

Works fine after upgrading to rc2 bits.

Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs]: I have added basic unit tests for SourceItemHandler.

**Pull request merged**